### PR TITLE
Argo upgrade

### PR DESCRIPTION
**Pull request by pzou1974**
Add workflowTemplate to crd
add workflowTeimpate to rbac

### CODE REVIEW

File: config/argo/argo.yaml [hunk 1]
```diff
@@ -126,62 +201,90 @@ roleRef:
   kind: ClusterRole
   name: argo-cluster-role
 subjects:
-  - kind: ServiceAccount
-    name: argo
-    namespace: system
+- kind: ServiceAccount
+  name: argo
+  namespace: argo
```
Comment (kevdowney) [inline config/argo/argo.yaml]: namespace needs to be `system`

File: config/argo/argo.yaml [hunk 2]
```diff
@@ -2,13 +2,23 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: argo
-  namespace: system
 ---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: argo-ui
-  namespace: system
```
Comment (kevdowney) [inline config/argo/argo.yaml]: namespace should be `system`

File: config/argo/argo.yaml [hunk 3]
```diff
@@ -2,13 +2,23 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: argo
-  namespace: system
```
Comment (kevdowney) [inline config/argo/argo.yaml]: namespace should be `system`

File: config/argo/argo.yaml [hunk 4]
```diff
@@ -126,62 +201,90 @@ roleRef:
...
+subjects:
+- kind: ServiceAccount
+  name: argo-ui
+  namespace: argo
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: workflow-controller-configmap
-  namespace: system
```
Comment (kevdowney) [inline config/argo/argo.yaml]: namespace needs to stay `system`

File: config/argo/argo.yaml [hunk 5]
```diff
@@ -126,62 +201,90 @@ roleRef:
...
-data:
-  config: |
-    namespace: addon-manager-system
```
Comment (kevdowney) [inline config/argo/argo.yaml]: `data` needs to be restored as-is

File: config/argo/argo.yaml [hunk 6]
```diff
@@ -126,62 +201,90 @@ roleRef:
...
+---
+apiVersion: v1
```
Comment (kevdowney) [inline config/argo/argo.yaml]: We don't deploy UI for Argo, should remove all `argo-ui`

File: config/argo/argo.yaml [hunk 7]
```diff
@@ -126,62 +201,90 @@ roleRef:
...
+        - name: ENABLE_WEB_CONSOLE
+          value: "false"
+        - name: BASE_HREF
+          value: /
+        image: argoproj/argoui:v2.4.1
+        name: argo-ui
+      serviceAccountName: argo-ui
 ---
-apiVersion: addonmgr.keikoproj.io/v1alpha1
-kind: Addon
```
Comment (kevdowney) [inline config/argo/argo.yaml]: kind `Addon` needs to be restored.

File: config/argo/argo.yaml (hunk 2 above)
Comment (pzou1974) [reply config/argo/argo.yaml]: added back.  thanks for review

**Pull request merged**